src/github.com/getlantern/fronted/server.go

```diff
@@ -169,27 +174,63 @@ func (server *Server) Serve(l net.Listener) error {
 // in a countingConn if an InstanceId was configured.
 func (server *Server) dialDestination(addr string) (net.Conn, error) {
 	if !server.AllowNonGlobalDestinations {
-		host, _, err := net.SplitHostPort(addr)
+		err := server.checkForLocalAddress(addr)
 		if err != nil {
-			err = fmt.Errorf("Unable to split host and port for %v: %v", addr, err)
 			log.Error(err.Error())
 			return nil, err
 		}
-		ipAddr, err := net.ResolveIPAddr("ip", host)
+	}
+
+	if server.AllowedPorts != nil {
+		err := server.checkForDisallowedPort(addr)
 		if err != nil {
-			err = fmt.Errorf("Unable to resolve destination IP addr: %s", err)
-			log.Error(err.Error())
-			return nil, err
-		}
-		if !ipAddr.IP.IsGlobalUnicast() {
-			err = fmt.Errorf("Not accepting connections to non-global address: %s", host)
 			log.Error(err.Error())
 			return nil, err
 		}
 	}
+
 	return net.DialTimeout("tcp", addr, dialTimeout)
 }
 
+func (server *Server) checkForLocalAddress(addr string) error {
+	host, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		return fmt.Errorf("Unable to split host and port for %v: %v", addr, err)
+	}
+
+	ipAddr, err := net.ResolveIPAddr("ip", host)
+	if err != nil || len(ipAddr.IP) == 0 {
```

Comment:
This was a drive-by fix.  I noticed that if passing in something like ":0", ResolveIPAddr would not return an error, but line 206 would fail because IP is empty.